### ISSUE TITLE
Query return different result when enable_pipeline_engine is true;

### ISSUE THREAD

**Issue by yongbingwang** [StarRocks/starrocks#2356]
### Steps to reproduce the behavior (Required)
```
CREATE TABLE IF NOT EXISTS t0 (c_0_0 INT NOT NULL ,c_0_1 CHAR(1) NULL ) UNIQUE KEY (c_0_0) DISTRIBUTED BY HASH (c_0_0) properties("replication_num"="1") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 BOOLEAN NULL ,c_1_1 DECIMAL64(15, 4) NULL ,c_1_2 DECIMAL(7, 3) NOT NULL ,c_1_3 DECIMAL128(14, 0) NULL ,c_1_4 CHAR(31) NOT NULL ,c_1_5 DECIMAL32(5, 2) NULL ,c_1_6 DECIMAL32(6, 0) NOT NULL ) DUPLICATE KEY (c_1_0,c_1_1,c_1_2,c_1_3,c_1_4) DISTRIBUTED BY HASH (c_1_1,c_1_2) properties("replication_num"="1") ;
CREATE TABLE IF NOT EXISTS t2 (c_2_0 VARCHAR(11) NULL ,c_2_1 DECIMAL64(13, 7) NULL ,c_2_2 CHAR(21) REPLACE_IF_NOT_NULL NOT NULL ) AGGREGATE KEY (c_2_0,c_2_1) DISTRIBUTED BY HASH (c_2_1) properties("replication_num"="1") ;
INSERT INTO t1 VALUES (true, 7939556.3011, 0.000, 53540670925274, "F", 9.27, 435029);
INSERT INTO t0 VALUES (24, ""), (28, ""), (24, "");
INSERT INTO t2 VALUES ("TDCnP73nE", 0.0099742, "nLsA"), (NULL, 0.0923564, "hh");
INSERT INTO t2 VALUES ("Jir", 0.0728920, "");
INSERT INTO t2 VALUES ("E9", 0.0060752, "u7q"), ("", 0.0713655, NULL);
INSERT INTO t1 VALUES (NULL, 4948928.4200, 7.630, 7945217223820, "f1", 0.88, 458850);
INSERT INTO t1 VALUES (true, NULL, 5.475, 72575734979739, "J8", 6.05, 152722), (false, 7442737.0955, 5.093, 66552004989694, "G", 8.13, 241004);
INSERT INTO t1 VALUES (true, NULL, 9.548, 44403720377629, "", 1.85, 619814), (false, 2766961.1179, 7.723, 91004592732992, "8qEo9", NULL, 210260);
INSERT INTO t0 VALUES (2, "");
mysql> set enable_pipeline_engine=true;
Query OK, 0 rows affected (0.01 sec)

mysql> SELECT t0.c_0_1, t2.c_2_0, t2.c_2_2 FROM t1, t2 LEFT JOIN t0 ON t2.c_2_0 = t0.c_0_1 AND t2.c_2_0 > "" WHERE (NOT ((NOT ((t0.c_0_1) IN ("") )))) GROUP BY t0.c_0_1, t2.c_2_0, t2.c_2_2;
+-------+-------+-------+
| c_0_1 | c_2_0 | c_2_2 |
+-------+-------+-------+
|       |       | NULL  |
+-------+-------+-------+
1 row in set (0.08 sec)

mysql> set enable_pipeline_engine=false;
Query OK, 0 rows affected (0.01 sec)

mysql> SELECT t0.c_0_1, t2.c_2_0, t2.c_2_2 FROM t1, t2 LEFT JOIN t0 ON t2.c_2_0 = t0.c_0_1 AND t2.c_2_0 > "" WHERE (NOT ((NOT ((t0.c_0_1) IN ("") )))) GROUP BY t0.c_0_1, t2.c_2_0, t2.c_2_2;
Empty set (0.07 sec)

```
### Expected behavior (Required)
```
mysql> SELECT t0.c_0_1, t2.c_2_0, t2.c_2_2 FROM t1, t2 LEFT JOIN t0 ON t2.c_2_0 = t0.c_0_1 AND t2.c_2_0 > "" WHERE (NOT ((NOT ((t0.c_0_1) IN ("") )))) GROUP BY t0.c_0_1, t2.c_2_0, t2.c_2_2;
Empty set (0.07 sec)
```
### Real behavior (Required)
```
mysql> SELECT t0.c_0_1, t2.c_2_0, t2.c_2_2 FROM t1, t2 LEFT JOIN t0 ON t2.c_2_0 = t0.c_0_1 AND t2.c_2_0 > "" WHERE (NOT ((NOT ((t0.c_0_1) IN ("") )))) GROUP BY t0.c_0_1, t2.c_2_0, t2.c_2_2;
+-------+-------+-------+
| c_0_1 | c_2_0 | c_2_2 |
+-------+-------+-------+
|       |       | NULL  |
+-------+-------+-------+
1 row in set (0.08 sec)
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - all

**Comment by satanson** [comment]
version: f4cef6c

**Comment by satanson** [comment]
https://github.com/StarRocks/starrocks/pull/2187
https://github.com/StarRocks/starrocks/pull/2141

**Comment by satanson** [comment]
```
mysql> select * from t0 where  c_0_1 > '' and c_0_1 = '';
+-------+-------+
| c_0_0 | c_0_1 |
+-------+-------+
|    24 |       |
|    28 |       |
|     2 |       |
+-------+-------+
3 rows in set (0.01 sec)

mysql> set enable_pipeline_engine=false;
Query OK, 0 rows affected (0.00 sec)

mysql> select * from t2 where  c_2_0 > '' and c_2_0 = '';
Empty set (0.01 sec)
```

**Comment by liuyehcf** [comment]
Fixed by [#2438](https://github.com/StarRocks/starrocks/pull/2438)
